contracts/cosmwasm-vm/cw-xcall/src/execute_call.rs

```diff
@@ -82,7 +82,7 @@ impl<'a> CwCallService<'a> {
             cosmwasm_std::SubMsgResult::Ok(_res) => {
                 let code = CallServiceResponseType::CallServiceResponseSuccess.into();
 
```

Comment:
Where is the message specific execution?

Reply:
on execution only case currently is whether  request needs response or not which is handled. other cases can also be added similarly without needing to know the message type explicitly.